README.md
seds-code
=========

code used for various chapter functions between civicrm, buddypress, and wordpress.


phpmailer/changelog.md
# ChangeLog

* Add Ukranian translation from @Krezalis
* Support for do_verp
* Fix bug in CRAM-MD5 AUTH
* Propagate Debugoutput option to SMTP class (@Reblutus)
* Determine MIME type of attachments automatically
* Add cross-platform, multibyte-safe pathinfo replacement (with tests) and use it
* Add a new 'html' Debugoutput type
* Clean up SMTP debug output, remove embedded HTML
* Some small changes in header formatting to improve IETF msglint test results
* Update test_script to use some recently changed features, rename to code_generator
* Generated code actually works!
* Update SyntaxHighlighter
* Major overhaul and cleanup of example code
* New PHPMailer graphic
* MsgHTML now uses RFC2392-compliant content ids
* Add line break normalization function and use it in MsgHTML
* Don't set unnecessary reply-to addresses
* Make fakesendmail.sh a bit cleaner and safer
* Set a content-transfer-encoding on multiparts (fixes msglint error)
* Fix cid generation in MsgHTML (Thanks to @digitalthought)
* Fix handling of multiple SMTP servers (Thanks to @NanoCaiordo)
* SMTP->Connect() now supports stream context options (Thanks to @stanislavdavid)
* Add support for iCal event alternatives (Thanks to @reblutus)
* Update to Polish language file (Thanks to Krzysztof Kowalewski)
* Update to Norwegian language file (Thanks to @datagutten)
* Update to Hungarian language file (Thanks to @dominicus-75)
* Add Persian/Farsi translation from @jaii

## Version 5.2.6 (April 11th 2013)
* Reflect move to PHPMailer GitHub organisation at https://github.com/PHPMailer/PHPMailer
* Fix unbumped version numbers
* Update packagist.org with new location
* Clean up Changelog

## Version 5.2.5 (April 6th 2013)
* First official release after move from Google Code
* Fixes for qmail when sending via mail()
* Merge in changes from Google code 5.2.4 release
* Minor coding standards cleanup in SMTP class
* Improved unit tests, now tests S/MIME signing
* Travis-CI support on GitHub, runs tests with fake SMTP server

## Version 5.2.4 (February 19, 2013)
* Fix tag and version bug.
* un-deprecate isSMTP(), isMail(), IsSendmail() and isQmail().
* Numerous translation updates

## Version 5.2.3 (February 8, 2013)
* Fix issue with older PCREs and ValidateAddress() (Bugz: 124)
* Add CRAM-MD5 authentication, thanks to Elijah madden, https://github.com/okonomiyaki3000
* Replacement of obsolete Quoted-Printable encoder with a much better implementation
* Composer package definition
* New language added: Hebrew

## Version 5.2.2 (December 3, 2012)
* Some fixes and syncs from https://github.com/Synchro/PHPMailer
* Add Slovak translation, thanks to Michal Tinka

## Version 5.2.2-rc2 (November 6, 2012)
* Fix SMTP server rotation (Bugz: 118)
* Allow override of autogen'ed 'Date' header (for Drupal's
  og_mailinglist module)
* No whitespace after '-f' option (Bugz: 116)
* Work around potential warning (Bugz: 114)

## Version 5.2.2-rc1 (September 28, 2012)
* Header encoding works with long lines (Bugz: 93)
* Turkish language update (Bugz: 94)
* undefined $pattern in EncodeQ bug squashed (Bugz: 98)
* use of mail() in safe_mode now works (Bugz: 96)
* ValidateAddress() now 'public static' so people can override the
  default and use their own validation scheme.
* ValidateAddress() no longer uses broken FILTER_VALIDATE_EMAIL
* Added in AUTH PLAIN SMTP authentication

## Version 5.2.2-beta2 (August 17, 2012)
* Fixed Postfix VERP support (Bugz: 92)
* Allow action_function callbacks to pass/use
  the From address (passed as final param)
* Prevent inf look for get_lines() (Bugz: 77)
* New public var ($UseSendmailOptions). Only pass sendmail()
  options iff we really are using sendmail or something sendmail
  compatible. (Bugz: 75)
* default setting for LE returned to "\n" due to popular demand.

## Version 5.2.2-beta1 (July 13, 2012)
* Expose PreSend() and PostSend() as public methods to allow
  for more control if serializing message sending.
* GetSentMIMEMessage() only constructs the message copy when
 needed. Save memory.
* Only pass params to mail() if the underlying MTA is
  "sendmail" (as defined as "having the string sendmail
  in its pathname") [#69]
* Attachments now work with Amazon SES and others [Bugz#70]
* Debug output now sent to stdout (via echo) or error_log [Bugz#5]
* New var: Debugoutput (for above) [Bugz#5]
* SMTP reads now Timeout aware (new var: Timeout=15) [Bugz#71]
* SMTP reads now can have a Timelimit associated with them
  (new var: Timelimit=30)[Bugz#71]
* Fix quoting issue associated with charsets
* default setting for LE is now RFC compliant: "\r\n"
* Return-Path can now be user defined (new var: ReturnPath)
  (the default is "" which implies no change from previous
  behavior, which was to use either From or Sender) [Bugz#46]
* X-Mailer header can now be disabled (by setting to a
  whitespace string, eg "  ") [Bugz#66]
* Bugz closed: #68, #60, #42, #43, #59, #55, #66, #48, #49,
               #52, #31, #41, #5. #70, #69

## Version 5.2.1 (January 16, 2012)
* Closed several bugs#5
* Performance improvements
* MsgHTML() now returns the message as required.
* New method: GetSentMIMEMessage() (returns full copy of sent message)

## Version 5.2 (July 19, 2011)
* protected MIME body and header
* better DKIM DNS Resource Record support
* better aly handling
* htmlfilter class added to extras
* moved to Apache Extras


leaderlist.php
<?php

add_action('bp_init', 'seds_chapterleaders1');

function seds_chapterleaders1() {
if ( class_exists( 'BP_Group_Extension' ) ) { // Recommended, to prevent problems during upgrade or when Groups are disabled
 //Run to find out if a chapter
 	
	global $bp;
 	$seds_ck_currgroup = $bp->groups->current_group->id;
 	if(is_numeric($seds_ck_currgroup)) { //Check if the group is set
 		global $wpdb;
 		$seds_ck_querytext = 'SELECT * FROM `wp_bpcivi_groupsync` WHERE `buddypress_group` =' . $seds_ck_currgroup;
		$seds_ck_settinggroups = $wpdb->get_results($seds_ck_querytext);
 	}
	
 //Function class call
    class SEDSLeadersedit extends BP_Group_Extension {
   	 function __construct() {
        	$this->name = 'Chapter Leadership';
            $this->slug = 'seds-leaders';
            $this->nav_item_position = 25;
            $this->visibility = 'public';
            //$this->enable_nav_item = false;
            $this->enable_create_step = false;
		}

	function display() {
    //Include Files
	include_once(ABSPATH  . '/wp-blog-header.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm.settings.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm/CRM/Core/Config.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm/civicrm.config.php');
	$config = CRM_Core_Config::singleton();
	global $wpdb;
	global $bp;
	//VARS
	
	//Query order of officers
		$seds_officerparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
		'option_sort'=>"weight",
		'option_group_id' => 88,);
		$seds_officerresult = civicrm_api('OptionValue', 'get', $seds_officerparams);
		$seds_officerlist = $seds_officerresult['values'];
	//Current Buddypress Group
		$seds_currgroup = $bp->groups->current_group->id;
	//Run Query on DB
		$seds_querytext = 'SELECT * FROM `wp_bpcivi_groupsync` WHERE `buddypress_group` =' . $seds_currgroup;
		$seds_settinggroups = $wpdb->get_results($seds_querytext);
	//Assign to array from first membership found - oldest set effectively
		$seds_groupsettings = get_object_vars($seds_settinggroups[0]);
	//Perform Query of Civicrm API
		$seds_leaderparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
			'contact_id_b' => $seds_groupsettings['orgid'],'relationship_type_id' => 5, //Employee Relationship
			'option_sort'=>"end_date",);
		$seds_leaderresult = civicrm_api('Relationship', 'get', $seds_leaderparams);
	//Cycle Through relationships and pull data from api call, assign data
	for ($i=0;$i<count($seds_leaderresult['values']);$i++) {
		$seds_leaderid[$i] = $seds_leaderresult['values'][$i]['contact_id_a'];
		$seds_leader_startdate[$i] = $seds_leaderresult['values'][$i]['start_date'];
		$seds_leader_enddate[$i] = $seds_leaderresult['values'][$i]['end_date'];
		$seds_leader_active[$i] = $seds_leaderresult['values'][$i]['is_active'];
		$seds_leader_title[$i] = $seds_leaderresult['values'][$i]['custom_3'];
		$seds_leader_titlel[$i] = $seds_leaderresult['values'][$i]['custom_3_1'];
		//$seds_leader_chapterrep[$i] = $seds_leaderresult['values'][$i]['custom_18'];
		//$seds_leader_chapterrepl[$i] = $seds_leaderresult['values'][$i]['custom_18_1'];
		//Pull Contact Data
			$seds_leadsparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
				'contact_id' => $seds_leaderid[$i],);
			$seds_leadsresult = civicrm_api('Contact', 'get', $seds_leadsparams);
		//Assign contact id info
		$seds_leader_displayname[$i] = $seds_leadsresult['values'][0]['display_name'];
		$seds_leader_addressid[$i] = $seds_leadsresult['values'][0]['address_id'];
		$seds_leader_emailid[$i] = $seds_leadsresult['values'][0]['email_id'];
		$seds_leader_email[$i] = $seds_leadsresult['values'][0]['email'];
	}
	//Find Years Covered
	$seds_mindate = date("Y",strtotime(min($seds_leader_startdate)));
	$seds_maxdate = date("Y",strtotime(max($seds_leader_enddate)));
	echo '<table id="seds-leadertable">';
	echo '<tr id="seds-leadertabletitlerow"><td id="seds-leadertabletitlerowlefttext">' . " Name </td><td> Role </td><td> Start Date </td><td> End Date</td><td> Email </td> </tr>";
	for ($j=0;$j<($seds_maxdate-$seds_mindate+1);$j++) { //Each Year
		$seds_leaderyear = $seds_mindate + $j;
		$seds_startyeardate = $seds_leaderyear . "-01-01";
		$seds_endyeardate = $seds_leaderyear . "-12-31";
		echo '<tr><td colspan="6" id="seds-leaderyear">' . $seds_leaderyear . "</td></tr>";
		for ($i=0;$i<count($seds_leaderresult['values']);$i++) {
			if( $seds_leader_startdate[$i] <=  $seds_endyeardate && $seds_leader_enddate[$i] >= $seds_startyeardate ) {
			if($seds_leaderresult['values'][$i]['custom_5'] == 0) {
				echo '<tr><td id="seds-leaderdisplayname">' . $seds_leader_displayname[$i] . "</td><td>";
				echo ucwords($seds_leader_title[$i]);
				echo "</td><td>";
				if($seds_leaderyear < date("Y",strtotime($seds_leader_enddate[$i]))) {
					echo $seds_leader_startdate[$i];
				}
				echo "</td><td>";
				if($seds_leaderyear > date("Y",strtotime($seds_leader_startdate[$i]))) {	
					echo $seds_leader_enddate[$i]; 
				}
				echo "</td><td>";
				echo $seds_leader_email[$i];
				echo "</td></tr>";
			}
			}
		}
	}
	echo "</table><br>";
}

/**
         * The content of the My Group Extension tab of the group admin
         */
	function edit_screen() {
	    if ( !bp_is_group_admin_screen( $this->slug ) )
			return false;
    //Include Files
	include_once(ABSPATH  . '/wp-blog-header.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm.settings.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm/CRM/Core/Config.php');
	include_once(ABSPATH  . 'wp-content/plugins/civicrm/civicrm/civicrm.config.php');
	$config = CRM_Core_Config::singleton();
	global $wpdb;
	global $bp;
	//Vars
	$actionurl = plugins_url() . "/seds-code/leaderlist/leader_post.php";
	$showhidejavaurl = plugins_url() . "/seds-code/leaderlist/showhide.js";
	$showhideeditjavaurl = plugins_url() . "/seds-code/leaderlist/showhideedit.js";
	$showhideeditjavaurlclose = plugins_url() . "/seds-code/leaderlist/showhideeditclose.js";

	//Query order of officers
		$seds_officerparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
		'option_sort'=>"weight",
		'option_group_id' => 88,);
		$seds_officerresult = civicrm_api('OptionValue', 'get', $seds_officerparams);
		$seds_officerlist = $seds_officerresult['values'];
	//Current Buddypress Group
		$seds_currgroup = $bp->groups->current_group->id;
	//Run Query on DB
		$seds_querytext = 'SELECT * FROM `wp_bpcivi_groupsync` WHERE `buddypress_group` =' . $seds_currgroup;
		$seds_settinggroups = $wpdb->get_results($seds_querytext);
	//Assign to array from first membership found - oldest set effectively
		$seds_groupsettings = get_object_vars($seds_settinggroups[0]);
	//Perform Query of Civicrm API
		$seds_leaderparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
			'contact_id_b' => $seds_groupsettings['orgid'],'relationship_type_id' => 5,
			'option_sort'=>"end_date",);
		$seds_leaderresult = civicrm_api('Relationship', 'get', $seds_leaderparams);
	//Cycle Through relationships and pull data from api call, assign data
	for ($i=0;$i<count($seds_leaderresult['values']);$i++) {
		$seds_relationid[$i] = $seds_leaderresult['values'][$i]['id'];
		$seds_leaderid[$i] = $seds_leaderresult['values'][$i]['contact_id_a'];
		$seds_leader_startdate[$i] = $seds_leaderresult['values'][$i]['start_date'];
		$seds_leader_enddate[$i] = $seds_leaderresult['values'][$i]['end_date'];
		$seds_leader_active[$i] = $seds_leaderresult['values'][$i]['is_active'];
		$seds_leader_title[$i] = $seds_leaderresult['values'][$i]['custom_3'];
		//$seds_leader_chapterrep[$i] = $seds_leaderresult['values'][$i]['custom_18'];
		//Pull Contact Data
			$seds_leadsparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
				'contact_id' => $seds_leaderid[$i],);
			$seds_leadsresult = civicrm_api('Contact', 'get', $seds_leadsparams);
		//Assign contact id info
		$seds_leader_displayname[$i] = $seds_leadsresult['values'][0]['display_name'];
		$seds_leader_addressid[$i] = $seds_leadsresult['values'][0]['address_id'];
		$seds_leader_emailid[$i] = $seds_leadsresult['values'][0]['email_id'];
		$seds_leader_email[$i] = $seds_leadsresult['values'][0]['email'];
		$seds_leader_customtitle[$i] = $seds_leadsresult['values'][0]['custom_4'];
	}
	//Find Years Covered
	$seds_mindate = date("Y",strtotime(min($seds_leader_startdate)));
	$seds_maxdate = date("Y",strtotime(max($seds_leader_enddate)));
	echo '<table id="seds-leadertable">';
	echo '<tr id="seds-leadertabletitlerow"><td id="seds-leadertabletitlerowlefttext">' . " Name </td><td> Role </td><td> ID </td><td> Start Date </td><td> End Date</td><td> Email </td><td> Edit / Delete </td></tr>";
	for ($j=0;$j<($seds_maxdate-$seds_mindate+1);$j++) { //Each Year
		$seds_leaderyear = $seds_mindate + $j;
		$seds_startyeardate = $seds_leaderyear . "-01-01";
		$seds_endyeardate = $seds_leaderyear . "-12-31";
		echo '<tr><td colspan="7" id="seds-leaderyear">' . $seds_leaderyear . "</td></tr>";
		for ($i=0;$i<count($seds_leaderresult['values']);$i++) { //Each Relationship
			if( $seds_leader_startdate[$i] <=  $seds_endyeardate && $seds_leader_enddate[$i] >= $seds_startyeardate ) {
			if($seds_leaderresult['values'][$i]['custom_5'] == 0) {
				echo '<tr><td id="seds-leaderdisplayname">' . $seds_leader_displayname[$i] . "</td><td>";
				if ($seds_leader_title[$i] == "project-group" || $seds_leader_title[$i] == "other") {
					echo ucwords($seds_leader_customtitle[$i]);
				} else {
					echo ucwords($seds_leader_title[$i]);
				}
				echo "</td><td>";
				echo $seds_leaderid[$i];
				echo "</td><td>";
				if($seds_leaderyear < date("Y",strtotime($seds_leader_enddate[$i]))) {
					echo $seds_leader_startdate[$i];
				}
				echo "</td><td>";
				if($seds_leaderyear > date("Y",strtotime($seds_leader_startdate[$i]))) {	
					echo $seds_leader_enddate[$i]; 
				}
				echo "</td><td>";
				echo $seds_leader_email[$i];
				echo "</td><td>";
				
				echo '<form action="'.$actionurl.'" method="post">';
					echo '<input type="hidden" name="seds_currgroup" value="' . $seds_currgroup . '">';
					echo '<input type="hidden" name="leaderredirect_url" value="' . get_site_url() . $_SERVER["REQUEST_URI"] . '">';
					echo '<input type="hidden" name="Edituser" value="' . $seds_leaderid[$i] . '">';
					echo '<input type="hidden" name="Relationedit" value="' . $seds_relationid[$i] . '">';
					echo '<input type="hidden" name="leadertitle" value="' . $seds_leader_title[$i] . '">';
					echo '<input type="hidden" name="leaderdname" value="' . $seds_leader_displayname[$i] . '">';
					echo '<input type="hidden" name="leaderstartdate" value="' . $seds_leader_startdate[$i] . '">';
					echo '<input type="hidden" name="leaderenddate" value="' . $seds_leader_enddate[$i] . '">';
					//echo '<input type="submit" id="'. "uredit-".$seds_relationid[$i].'" name="' . $seds_leaderid[$i] . '" value="' . "Edit" . '">';
					echo '<input type="button" id="'. "uredit-".$seds_relationid[$i].'" onclick="showdiv(\''. "ureditdiv-".$seds_relationid[$i] .'\');" name="' . $seds_leaderid[$i] . '" value="' . "Modify" . '">';
				echo "</form>";
				//End of Edit form
				
				//Insert hidden div
					echo '<div id="'."ureditdiv-".$seds_relationid[$i].'" style="display: none;">';
					echo '<div class="leaderblocker" style="position: absolute;left: 0px;top: 0px;background-color: rgba(255, 255, 255, 0.53);width: 10000px;height: 10000px;z-index: 9998;"></div>'; //Background div
					echo '<div class="leaderinput" style="position: absolute;left: 50px;z-index: 9999;background-color: rgba(43, 43, 43, 0.85);border-color: black;border-width: 2px;border-style: solid;border-radius: 10px;" class="leadereditboxes;">';
					echo '<input type="button" id="'. "ureditclose-".$seds_relationid[$i].'" onclick="hidediv1(\''. "ureditdiv-".$seds_relationid[$i] .'\');" value="' . "X" . '" style="float: right;">';
					echo '<table>';
					echo '<form action="'.$actionurl.'" method="post">';
					echo '<input type="hidden" name="leaderredirect_url" value="' . get_site_url() . $_SERVER["REQUEST_URI"] . '">';
					echo '<input type="hidden" name="seds_currgroup" value="' . $seds_currgroup . '">';
					echo '<tr class="seds-leadertabletitlerow">';
					echo "<tr><td> Name </td><td> Role </td><td> Start Date </td><td> End Date </td><td> </td>";
					echo "</tr>";
					echo "<tr><td>";
					echo $seds_leader_displayname[$i];
					echo "</td><td>";
					echo ucwords($seds_leader_title[$i]);
					echo "</td><td>";
					echo '<input type="date" name="changebday" value="' . $seds_leader_startdate[$i] . '">';
					echo "</td><td>";
					echo '<input type="date" name="changeendday" value="' . $seds_leader_enddate[$i] . '">';
					echo "</td><td>";
					echo '<input type="hidden" name="Changeuser" value="' . $seds_leaderid[$i] . '">';
					echo '<input type="hidden" name="Relationedit" value="' . $seds_relationid[$i] . '">';
					wp_nonce_field("sedsleaderedit", "_sedsleaderedit");
					echo '<input type="submit" name="' . "leaderChangesubmit" . '" value="' . "Change" . '">';
					echo '<input type="submit" name="' . "removeleaderDelete" . '" value="' . "Delete" . '">';
					echo "</td></tr>";
					echo '</form>';
					echo "</table>";
					echo "</div>"; //End Table div
					echo "</div>"; //End Hide Div
				
				//End of main table row
				echo "</td></tr>";
			}
			} //End of show Loop
		}
	}
	echo "</table><br></form>";
	echo '<script type="text/javascript" src="'.$showhideeditjavaurl.'"></script>';
	echo '<script type="text/javascript" src="'.$showhideeditjavaurlclose.'"></script>';
	//Pull Custom Dropdowns
	$seds_leadershipdropparams = array('version' => 3,'page' => 'CiviCRM','q' => 'civicrm/ajax/rest','sequential' => 1,
		'option_sort'=>"weight",
		'option_group_id' => 88,);
	$seds_leadershipdropresult = civicrm_api('OptionValue', 'get', $seds_leadershipdropparams);
	$seds_leadershipdrop = $seds_leadershipdropresult['values'];
	//Form for adding relationships / officers
	echo '<form action="'.$actionurl.'" method="post">';
	echo '<input type="hidden" name="leaderredirect_url" value="' . get_site_url() . $_SERVER["REQUEST_URI"] . '">';
	echo '<input type="hidden" name="seds_currgroup" value="' . $seds_currgroup . '">';
	echo "<table>";
	echo '<tr><td colspan="2"><h3>New Leadership</h3></td></tr>';
	echo '<tr><td> I.D. </td><td><input type="number" name="membid" required> </td></tr>';
	echo '<tr><td> Role </td><td>';
	echo '<select name="nrole" required>';
	for($z=0;$z<count($seds_leadershipdrop);$z++) {
		if($seds_leadershipdrop[$z]['is_active'] = 1) {
			echo '<option value="' . $seds_leadershipdrop[$z]['value'] . '">' . $seds_leadershipdrop[$z]['label'] . '</option>';
		}
	}
	echo '</select>';
	echo '</td></tr>';
	echo '<tr><td> Other/Project Group </td><td><input type="text" name="ncustom_4"> </td></tr>';
	//echo '<tr><td> Chapter Representative </td><td><input type="radio" name="nchapterrep" id="1" value="1"> Yes <br> <input type="radio" name="nchapterrep" id="2" value="0"> No</td></tr>';
	echo '<tr><td> Start Date </td><td><input type="date" name="newstartdate" required></td></tr>';
	echo '<tr><td> End Date </td><td><input type="date" name="newenddate" required></td></tr>';
	echo "</table>";
	echo '<input type="hidden" name="nChapterid" value="' . $seds_groupsettings['orgid'] . '">';
	echo '<input type="hidden" name="leaderredirect_url" value="' . get_site_url() . $_SERVER["REQUEST_URI"] . '">';
	echo '<input type="hidden" name="seds_currgroup" value="' . $seds_currgroup . '">';
	wp_nonce_field("sedsleaderedit", "_sedsleaderedit");
	echo '<input type="submit" name="NLeadersubmit" value="' . "New Leader" . '">';
	echo "</form>";
	
	//Deleted Members Table
	//JS Button
	echo '<input type="button" id="hideshowleaders" value="Hide/Show Deleted">';
	//Hide and show this div
	echo '<div id="sedsleadertabledels" style="display: none">';
	echo '<table id="seds-leadertable">';
	echo '<tr id="seds-leadertabletitlerow"><td id="seds-leadertabletitlerowlefttext">' . " Name </td><td> Role </td><td> ID </td><td> Start Date </td><td> End Date</td><td> Email </td><td> Edit / Delete </td></tr>";
	for ($j=0;$j<($seds_maxdate-$seds_mindate+1);$j++) { //Each Year
		$seds_leaderyear = $seds_mindate + $j;
		$seds_startyeardate = $seds_leaderyear . "-01-01";
		$seds_endyeardate = $seds_leaderyear . "-12-31";
		echo '<tr><td colspan="7" id="seds-leaderyear">' . $seds_leaderyear . "</td></tr>";
		for ($i=0;$i<count($seds_leaderresult['values']);$i++) { //Each Relationship
			if( $seds_leader_startdate[$i] <=  $seds_endyeardate && $seds_leader_enddate[$i] >= $seds_startyeardate ) {
			if($seds_leaderresult['values'][$i]['custom_5'] == 1) {
				echo '<tr><td id="seds-leaderdisplayname">' . $seds_leader_displayname[$i] . "</td><td>";
				echo ucwords($seds_leader_title[$i]);
				echo "</td><td>";
				echo $seds_leaderid[$i];
				echo "</td><td>";
				if($seds_leaderyear < date("Y",strtotime($seds_leader_enddate[$i]))) {
					echo $seds_leader_startdate[$i];
				}
				echo "</td><td>";
				if($seds_leaderyear > date("Y",strtotime($seds_leader_startdate[$i]))) {	
					echo $seds_leader_enddate[$i]; 
				}
				echo "</td><td>";
				echo $seds_leader_email[$i];
				echo "</td><td>";
				
				echo '<form action="'.$actionurl.'" method="post">';
					echo '<input type="hidden" name="leaderredirect_url" value="' . get_site_url() . $_SERVER["REQUEST_URI"] . '">';
					echo '<input type="hidden" name="seds_currgroup" value="' . $seds_currgroup . '">';
					echo '<input type="hidden" name="Edituser" value="' . $seds_leaderid[$i] . '">';
					echo '<input type="hidden" name="Changeuser" value="' . $seds_leaderid[$i] . '">';
					echo '<input type="hidden" name="Relationedit" value="' . $seds_relationid[$i] . '">';
					echo '<input type="hidden" name="Undeleteuser1" value="' . $seds_leaderid[$i] . '">';
					echo '<input type="hidden" name="Relationundel" value="' . $seds_relationid[$i] . '">';
					echo '<input type="hidden" name="leadertitle" value="' . $seds_leader_title[$i] . '">';
					echo '<input type="hidden" name="leaderdname" value="' . $seds_leader_displayname[$i] . '">';
					wp_nonce_field("sedsleaderedit", "_sedsleaderedit");
					echo '<input type="submit" name="' . "removeleaderDelete" . '" value="' . "UnDelete" . '">';
					echo '<input type="submit" name="' . "PermDelete" . '" value="' . "PermDelete" . '">';
				echo "</form>";
				echo "</td></tr>";
			}
			} //End of Deleted Loop
		}
	}
	echo "</table><br></form>";
	echo "</div>";
	echo '<script type="text/javascript" src="'.$showhidejavaurl.'"></script>';
	//}
	
	/*//Diagnostics
	echo "Post: <pre>";
	print_r($_POST);
	echo "</pre>";
	
	echo "Relationships: <pre>";
	print_r($seds_leaderresult);
	echo "</pre>"; */

} //End of Edit Screen

} //End of Class
	if(count($seds_ck_settinggroups) > 0) { //Make it so that the group exension is only used for chapter
		bp_register_group_extension( 'SEDSLeadersedit' );
	}
}
}
